marimo/_plugins/ui/_impl/table.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2023 Marimo. All rights reserved.
+from __future__ import annotations
+
+from typing import (
+    TYPE_CHECKING,
+    Callable,
+    Final,
+    List,
+    Literal,
+    Optional,
+    Sequence,
+    Union,
+    cast,
+)
+
+from marimo import _loggers
+from marimo._output.mime import MIME
+from marimo._output.rich_help import mddoc
+from marimo._plugins.ui._core.ui_element import UIElement
+
+LOGGER = _loggers.marimo_logger()
+
+Numeric = Union[int, float]
+
+if TYPE_CHECKING:
+    import pandas as pd
+
+TableData = Union[
+    list[dict[str, str]],
+    Sequence[str | int | float | bool | MIME | None],
```

Comment:
Why do we need `list[dict[str, str]]`?

Reply:
technically our values in the dicts need to be homegenious. this makes the typings super ugly, so i removed this and just type ignored the response from `to_dicts`